src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpServer.cs

```diff
@@ -77,12 +119,17 @@ private static NativeMethods.REQUEST_NOTIFICATION_STATUS HandleRequest(IntPtr pI
         {
             // Unwrap the server so we can create an http context and process the request
             var server = (IISHttpServer)GCHandle.FromIntPtr(pvRequestContext).Target;
+            if (server != null)
+            {
+                Interlocked.Increment(ref server._outstandingRequests);
+            }
 
             var context = server._iisContextFactory.CreateHttpContext(pInProcessHandler);
 
             var task = Task.Run(() => context.ProcessRequestAsync());
 
-            task.ContinueWith((t, state) => CompleteRequest((IISHttpContext)state, t), context);
+            // TODO make this use continuations correctly instead of closures.
+            task.ContinueWith((t, state) => CompleteRequest(context, (IISHttpServer)state, t), server);
```

Comment:
You just introduced a closure. I'd rather you store a reference to the IISHttpServer on the IISHttpContext